recipes/default.rb

```diff
@@ -0,0 +1,126 @@
+# Copyright:: 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Validate the agent type
+unless %w(google-cloud-ops ops-agent monitoring logging).include? node['final_agent_type']
+  Chef::Log.fatal("Received invalid agent type: '#{node['agent_type']}'. The Cloud Ops Chef cookbook supports the following agents: 'monitoring', 'logging' and 'ops-agent'.")
+end
+# For windows, only the ops-agent is supported
+if platform_family?('windows') && (%w(monitoring logging).include? node['final_agent_type'])
```

Comment:
Curious: Why do we need to have both `node['final_agent_type']` and `node['agent_type']`?

Reply:
This is in line with both Ansible and Puppet projects as well, but done as a permanent variable rather than in line several places throughout the code. As mentioned in the other comment, `ops-agent` is specified by the customer, but `google-cloud-ops` is what is needed for file and service names.

Reply:
Ah I see. To help with readability for future developers, can we delay the transformation to later stage (aka after the agent type validation and before constructing the service name and installation script name). Similar to how [Puppet](https://github.com/GoogleCloudPlatform/google-cloud-ops-agents-puppet/blob/ff1be4759a86a4a46c8f2c5b7bf961fce017f0ec/manifests/agent.pp#L47) and [Ansible](https://github.com/GoogleCloudPlatform/google-cloud-ops-agents-ansible/blob/17df6d63d560bccf76ccbef01c781caf39d805a7/tasks/linux.yml#L13) handle it. (Puppet does not use the installation script, so it does not have similar issue for the installation script.)

Reply:
I've got code in a branch for this, I'll be having @jsirianni review it soon so it can be pushed to our master and included into this PR.